userspace/libsinsp/filterchecks.cpp

```diff
@@ -109,6 +110,317 @@ static inline bool str_match_start(const std::string& val, size_t len, const cha
+///////////////////////////////////////////////////////////////////////////////
+// sinsp_filter_check_fspath implementation
+///////////////////////////////////////////////////////////////////////////////
+const filtercheck_field_info sinsp_filter_check_fspath_fields[] =
+	{
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.name", "Path for Filesystem-related operation", "For any event type that deals with a filesystem path, the path the file syscall is operating on. This path is always fully resolved, prepending the thread cwd when needed."},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.nameraw", "Raw path for Filesystem-related operation", "For any event type that deals with a filesystem path, the path the file syscall is operating on. This path is always the path provided to the syscall and may not be fully resolved."},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.source", "Source path for Filesystem-related operation", "For any event type that deals with a filesystem path, and specifically for a source and target like mv, cp, etc, the source path the file syscall is operating on. This path is always fully resolved, prepending the thread cwd when needed."},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.sourceraw", "Source path for Filesystem-related operation", "For any event type that deals with a filesystem path, and specifically for a source and target like mv, cp, etc, the source path the file syscall is operating on. This path is always the path provided to the syscall and may not be fully resolved."},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.target", "Target path for Filesystem-related operation", "For any event type that deals with a filesystem path, and specifically for a target and target like mv, cp, etc, the target path the file syscall is operating on. This path is always fully resolved, prepending the thread cwd when needed."},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "fspath.targetraw", "Target path for Filesystem-related operation", "For any event type that deals with a filesystem path, and specifically for a target and target like mv, cp, etc, the target path the file syscall is operating on. This path is always the path provided to the syscall and may not be fully resolved."},
+};
+
+sinsp_filter_check_fspath::sinsp_filter_check_fspath()
+{
+	m_info.m_name = "file";
+	m_info.m_desc = "Every syscall that has a filesystem path in its arguments has these fields set with information related to the path arguments. This differs from the fd.* fields as it includes syscalls like unlink, rename, etc. that act directly on filesystem paths as compared to opened file descriptors.";
+	m_info.m_fields = sinsp_filter_check_fspath_fields;
+	m_info.m_nfields = sizeof(sinsp_filter_check_fspath_fields) / sizeof(sinsp_filter_check_fspath_fields[0]);
+	m_info.m_flags = filter_check_info::FL_WORKS_ON_THREAD_TABLE;
+};
+
+std::shared_ptr<gen_event_filter_check> sinsp_filter_check_fspath::create_check(const char *name,
```

Comment:
From a design perspective, we're adding yet one more tedious circular dependency. We have the filter factory, which depends on the filterchecks definitions, and then now also one of the filterchecks defintions depending on the filter factory. Wouldn't it be easier at this level to create the filterchecks manually? Most of the ones you use are `evt.*` anyways, so it will be few lines of code for allocating the right filtercheck class and avoid tangling the code dependencies even more than what we already have.

Reply:
Yeah, I can get rid of the factory and just create the `evt.arg` and `fd.*` checks directly.